src/service/chat.js

```diff
@@ -206,6 +206,71 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const getAllowAnonDmStatus = async (postId) => {
+  try {
+    const response = await api.get(`/users/check-allow-dm/?source=post&post_id=${postId}`);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getAllowAnonDmStatus` function is well written and handles errors properly. However, it's recommended to encode URL parameters to ensure they are safe to be used within a URL. This can prevent potential security issues related to URL injection.
```diff
- const response = await api.get(`/users/check-allow-dm/?source=post&post_id=${postId}`);
+ const encodedPostId = encodeURIComponent(postId);
+ const response = await api.get(`/users/check-allow-dm/?source=post&post_id=${encodedPostId}`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->